packages/ui-extensions-react/src/surfaces/customer-account/hooks/app-metafields.ts

```diff
@@ -25,7 +25,7 @@ type AppMetafieldFilterKeys = keyof AppMetafieldFilters;
  * @arg {AppMetafieldFilters} - filter the list of returned metafields
  */
 export function useAppMetafields<
-  Target extends RenderExtensionTarget = RenderExtensionTarget,
+  Target extends OrderStatusExtensionTarget = OrderStatusExtensionTarget,
```

Comment:
Reason: `appMetafields` only exists on `OrderStatusApi` instead of other APIs. Only extensions on `OrderStatus` can call this hook. Thus, we should limit the targets to the targets on OrderStatus